### PR TITLE
Remove NR Session when user role changes

### PR DESCRIPTION
**Pull request by hardillb**
fixes https://github.com/FlowFuse/security/issues/82

## Description

Ensure that any open NR sessions for the user are removed so they no longer have access

Need to work out test cases for this.

## Related Issue(s)

https://github.com/FlowFuse/security/issues/82

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Review by knolleary — CHANGES_REQUESTED** [review]
Needs some tests, but otherwise looks okay

**Comment by hardillb** [comment]
Need to work out how to test this and get it merged

File: forge/db/controllers/Team.js [hunk 1]
```diff
@@ -90,8 +90,9 @@ module.exports = {
             }
             await userRole.destroy()
 
+            await app.db.controllers.StorageSession.removeUserFromTeamSessions(user, team)
```
Comment (ZJvandeWeg) [inline forge/db/controllers/Team.js:93]: Wouldn't the API: `await team.removeUserFromSessions(user)` be nicer, and make the model fat and keep the controller skinny?

https://medium.com/marmolabs/skinny-models-skinny-controllers-fat-services-e04cfe2d6ae

Comment (hardillb) [reply forge/db/controllers/Team.js:93]: We already have similar code in the controller, this keeps it all in the same place

**Review by knolleary — APPROVED** [review]
Noting ZJ's comment about fat controllers - we have some general work to do to tidy up what belongs in the controllers and what in the models. We will have to be opportunistic to do that work. For this PR, I'm okay with adding this functionality to the existing controller alongside the other session invalidation logic for now in order to keep things moving forward.

**Pull request merged**